BLE Indoor Positioning/src/test/java/com/nexenio/bleindoorpositioning/ble/beacon/BeaconCreator.java

```diff
@@ -0,0 +1,22 @@
+package com.nexenio.bleindoorpositioning.ble.beacon;
+
+import com.nexenio.bleindoorpositioning.ble.advertising.AdvertisingPacket;
+import com.nexenio.bleindoorpositioning.ble.advertising.IBeaconAdvertisingPacket;
+import com.nexenio.bleindoorpositioning.testUtil.AbstractBeaconCreator;
+
+public class BeaconCreator<B extends Beacon> extends AbstractBeaconCreator<B> {
+
+    public BeaconCreator(Class<B> beaconClass) {
+        super(beaconClass);
+    }
+
+    @Override
+    public AdvertisingPacket getAdvertisingPacketForBeaconClass(Class<B> beaconClass) throws InstantiationException {
+        String beaconClassName = beaconClass.getSimpleName();
+        if (beaconClassName.equals(IBeacon.class.getSimpleName())) {
```

Comment:
Use `beaconClass == IBeacon.class` instead of relying on the name (could be the same name in a different package for a different class).